partitions.go

```diff
@@ -111,6 +113,19 @@ func (p *partitions) updateLocalPartitions(local map[int]bool) {
 	}
 }
 
+func (p *partitions) deDupe(nodes []string) []string {
```

Comment:
also there's no reason for it to be on `p`: it can just be `func dedupe(s []string) []string`